xray/commands/audit/sca/java/deptreemanager.go

```diff
@@ -58,45 +59,39 @@ type depTreeNode struct {
 	Children []string `json:"children"`
 }
 
-// getGraphFromDepTree reads the output files of the gradle-dep-tree and maven-dep-tree plugins and returns them as a slice of GraphNodes.
+// Reads the output files of the gradle-dep-tree and maven-dep-tree plugins and returns them as a slice of GraphNodes.
 // It takes the output of the plugin's run (which is a byte representation of a list of paths of the output files, separated by newlines) as input.
 func getGraphFromDepTree(outputFilePaths string) (depsGraph []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
 	modules, err := parseDepTreeFiles(outputFilePaths)
 	if err != nil {
 		return
 	}
+
 	uniqueDepsSet := datastructures.MakeSet[string]()
-	for _, moduleTree := range modules {
-		directDepId := GavPackageTypeIdentifier + moduleTree.Root
-		directDependency := &xrayUtils.GraphNode{
-			Id:    directDepId,
-			Nodes: []*xrayUtils.GraphNode{},
+	for _, module := range modules {
+		moduleTreeMap := make(map[string][]string)
+		moduleDeps := module.Nodes
+		for depName, dependency := range moduleDeps {
+			dependencyId := GavPackageTypeIdentifier + depName
+			var childrenList []string
+			for _, childName := range dependency.Children {
+				childId := GavPackageTypeIdentifier + childName
+				childrenList = append(childrenList, childId)
+			}
+			if len(childrenList) > 0 {
+				moduleTreeMap[dependencyId] = childrenList
+			}
+		}
+		moduleTree, moduleUniqueDeps := sca.BuildXrayDependencyTree(moduleTreeMap, GavPackageTypeIdentifier+module.Root)
+		depsGraph = append(depsGraph, moduleTree)
+		for _, depToAdd := range moduleUniqueDeps {
+			uniqueDepsSet.Add(depToAdd)
 		}
```

Comment:
The Set in this case becomes irrelevant after your changes because we already obtain 'uniqueDeps' as a slice of strings. Therefore, we can skip initializing a set and directly convert it to a slice of strings (line 91).